packages/backend-for-frontend/src/app.ts

```diff
@@ -39,68 +35,95 @@ import delegationRouter from "./routers/delegationRouter.js";
+  authorizationService: AuthorizationService;
+  authorizationServiceForSupport: AuthorizationService;
+  catalogService: CatalogService;
+  clientService: ClientService;
+  delegationService: DelegationService;
+  eServiceTemplateService: EServiceTemplateService;
+  producerKeychainService: ProducerKeychainService;
+  purposeService: PurposeService;
+  selfcareService: SelfcareService;
+  tenantService: TenantService;
+  toolsService: ToolsService;
+};
 
-const fileManager = initFileManager(config);
-const allowList = await getAllowList(serviceName, fileManager, config);
+export type RateLimiterMiddleware = ReturnType<
+  typeof rateLimiterMiddlewareBuilder
+>;
 
-const clients = getInteropBeClients();
+export const serviceName = modelsServiceName.BACKEND_FOR_FRONTEND;
 
-const app = zodiosCtx.app();
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export async function createApp(
+  services: BFFServices,
+  rateLimiterMiddleware: RateLimiterMiddleware
+) {
+  const app = zodiosCtx.app();
 
-const redisRateLimiter = await initRedisRateLimiter({
-  limiterGroup: "BFF",
-  maxRequests: config.rateLimiterMaxRequests,
-  rateInterval: config.rateLimiterRateInterval,
-  burstPercentage: config.rateLimiterBurstPercentage,
-  redisHost: config.rateLimiterRedisHost,
-  redisPort: config.rateLimiterRedisPort,
-  timeout: config.rateLimiterTimeout,
-});
+  // Disable the "X-Powered-By: Express" HTTP header for security reasons.
+  // See https://cheatsheetseries.owasp.org/cheatsheets/HTTP_Headers_Cheat_Sheet.html#recommendation_16
+  app.disable("x-powered-by");
 
-// Disable the "X-Powered-By: Express" HTTP header for security reasons.
-// See https://cheatsheetseries.owasp.org/cheatsheets/HTTP_Headers_Cheat_Sheet.html#recommendation_16
-app.disable("x-powered-by");
+  app.disable("etag");
 
-app.disable("etag");
+  // parse files from multipart/form-data and put them in req.body
+  app.use(multerMiddleware);
+  app.use(fromFilesToBodyMiddleware);
 
-// parse files from multipart/form-data and put them in req.body
-app.use(multerMiddleware);
-app.use(fromFilesToBodyMiddleware);
+  // parse application/x-www-form-urlencoded and put it in req.body
+  app.use(express.urlencoded({ extended: true }));
 
-// parse application/x-www-form-urlencoded and put it in req.body
-app.use(express.urlencoded({ extended: true }));
+  app.use(loggerMiddleware(serviceName));
 
-app.use(loggerMiddleware(serviceName));
+  app.use(
+    `/backend-for-frontend/${config.backendForFrontendInterfaceVersion}`,
+    healthRouter,
+    contextMiddleware(serviceName, false),
+    await applicationAuditBeginMiddleware(serviceName, config),
+    await applicationAuditEndMiddleware(serviceName, config),
+    await applicationAuditEndSessionTokenExchangeMiddleware(
+      serviceName,
+      config
+    ),
+    authenticationMiddleware(config),
+    uiAuthDataValidationMiddleware(),
+    rateLimiterMiddleware
+  );
+  app.use(
```

Comment:
this should be restored to a single `app.use` block so that routers listen under the base path `/backend-for-frontend/${config.backendForFrontendInterfaceVersion}`

also please restore the comment line "Authenticated routes - rate limiter relies on auth data to work"

Reply:
also, instead of using `/backend-for-frontend/${config.backendForFrontendInterfaceVersion}` literally at line 96 (and then we'll have to reuse it in the tests), we can define it as done in PR https://github.com/pagopa/interop-be-monorepo/pull/1823/files in a file `[config/appBasePath.ts](https://github.com/pagopa/interop-be-monorepo/pull/1823/files#diff-29d3ef066553a75a82395d0396a5ed26cd0f4d1d68deacf86e0a85466d8f5120)` so we can import it here and in the tests